backend/userauths/models.py

```diff
@@ -1,3 +1,75 @@
 from django.db import models
+from django.contrib.auth.models import AbstractUser
+
+class User(AbstractUser):
+    """_summary_
+
+    Args:
+        AbstractUser (_type_): _description_
+
+    Returns:
+        _type_: _description_
+    """
+    username = models.CharField(unique=True, max_length=100)
+    email = models.EmailField(unique=True)
+    full_name = models.CharField(unique=True, max_length=100)
+    otp = models.CharField(max_length=100, null=True, blank=True)
+    refresh_token = models.CharField(max_length=1000, null=True, blank=True)
+
+    USERNAME_FIELD = 'email'
+    REQUIRED_FIELDS = ['username']
+
+    def __str__(self):
+        """_summary_
+
+        Returns:
+            _type_: _description_
+        """
+        return self.email
+    
+    def save(self, *args, **kwargs):
+        """_summary_
+        """
+        email_username, full_name = self.email.split("@")
+        if self.full_name == "" or self.full_name == None:
+            self.full_name == email_username
+        if self.username == "" or self.username == None:
+            self.username = email_username
+        super(User, self).save(*args, **kwargs)
+   
+class Profile(models.Model):
```

Comment:
Your class will only create user but not profile! Example you may have 500 users in the future and will you be adding those 500 users manually? Make use of **django.core.signals** to define new function that will handle profile creation and storing.